ui-trackoccupancydiagram/src/components/helpers/drawElements/drawOccupancyZones.ts

```diff
@@ -4,58 +4,197 @@ import {
   CANVAS_PADDING,
   OCCUPANCY_ZONE_START,
   OCCUPANCY_ZONE_HEIGHT,
+  FONTS,
   COLORS,
 } from '../../consts';
 import type { OccupancyZone, Track } from '../../types';
 
+const { SANS } = FONTS;
+const { REMAINING_TRAINS_BACKGROUND, WHITE_100, SELECTION_20 } = COLORS;
+const REMAINING_TRAINS_WIDTH = 70;
+const REMAINING_TRAINS_HEIGHT = 24;
+const REMAINING_TEXT_OFFSET = 12;
+const Y_OFFSET_INCREMENT = 4;
+const MAX_ZONES = 9;
+const X_BACKGROUND_OFFSET = 22;
+const X_TROUGHTRAIN_PADDING = 4;
+const Y_BACKGROUND_OFFSET = 20;
+const SELECTED_TRAIN_ID_GRADIANT = 2;
+
+type DrawZone = {
+  ctx: CanvasRenderingContext2D;
+  arrivalTime: number;
+  departureTime: number;
+  yPosition: number;
+};
+
+const drawDefaultZone = ({ ctx, arrivalTime, departureTime, yPosition }: DrawZone) => {
+  ctx.beginPath();
+  ctx.rect(arrivalTime, yPosition, departureTime! - arrivalTime, OCCUPANCY_ZONE_HEIGHT);
+  ctx.fill();
+  ctx.stroke();
+};
+
+const ARROW_OFFSET_X = 1;
+const ARROW_OFFSET_Y = 1.5;
+const ARROW_WIDTH = 4.5;
+const ARROW_TOP_Y = 3.5;
+const ARROW_BOTTOM_Y = 6.5;
+
+const drawThroughTrain = ({ ctx, arrivalTime }: Omit<DrawZone, 'departureTime' | 'yPosition'>) => {
```

Comment:
Maybe you can add some comments like this
```suggestion
// Through trains are materialized by an arrow like the following one
//  ___
//  \_/
//  / \
//  ‾‾‾
const drawThroughTrain = ({ ctx, arrivalTime }: Omit<DrawZone, 'departureTime' | 'yPosition'>) => {
```

And then: 
```
  // draw the upper part
  // draw the lower part
  // draw the white line in the middle
```

Reply:
done